tests/suites/casper/visible.js

```diff
@@ -1,16 +1,27 @@
 /*eslint strict:0*/
-casper.test.begin('visibility tests', 6, function(test) {
+casper.test.begin('visibility tests', 11, function(test) {
     casper.start('tests/site/visible.html', function() {
         test.assert(!this.visible('#img1'), 'Casper.visible() can detect if an element is invisible');
         test.assert(this.visible('#img2'), 'Casper.visible() can detect if an element is visible');
         test.assert(!this.visible('#img3'), 'Casper.visible() can detect if an element is invisible');
         test.assert(this.visible('#img4'), 'Casper.visible() can detect if an element with display:flex is visible');
         test.assert(this.visible('img'), 'Casper.visible() can detect if an element is visible');
-        this.waitWhileVisible('#img1', function() {
+        test.assert(!this.visible('#button1'), 'Casper.visible() can detect if an element is invisible when parent has display none');
+        test.assert(this.visible('#button2'), 'Casper.visible() can detect if an element is visible when parent is visible');
+        //test.assert(!this.visible('#button3'), 'Casper.visible() can detect if an element is invisible when parent has visibility hidden');//currently this case is not handled
+        test.assert(this.visible('#circle1'), 'Casper.visible() can detect if an element inside svg is visible');
+        test.assert(!this.visible('#circle2'), 'Casper.visible() can detect if an element inside svg is invisible when parent has display none');
+        //test.assert(!this.visible('#circle3'), 'Casper.visible() can detect if an element inside svg is invisible when parent has visibility hidden');//currently this case is not handled
```

Comment:
ok, will do that. Though I wrote those tests in case of future implementation will handle the case of a parent with `visibility: hidden`